### ISSUE TITLE
would you be interested in adding the official image of the coin to the list?

### ISSUE THREAD

**Issue by jpita** [KrauseFx/crypto-summaries#4]
would you be interested in adding the official image of the coin to the list?

**Comment by KrauseFx** [comment]
Interesting - nice idea 👍 It seems like the image isn't included in the [coinmarketcap API](https://coinmarketcap.com/api/) either, right? 

I wonder how this would work with image licenses though

**Comment by jpita** [comment]
I was thinking about some reliable sources (either the official site from the crypto or something like wikipedia).
No idea about the licenses though

**Comment by KrauseFx** [comment]
We could always just return the image URL, which right now would be

```
https://files.coinmarketcap.com/static/img/coins/32x32/ethereum.png
```
or larger
```
https://files.coinmarketcap.com/static/img/coins/64x64/ethereum.png
```

**Comment by jpita** [comment]
that would do the trick



<https://mailtrack.io/> Sent with Mailtrack
<https://mailtrack.io/install?source=signature&lang=en&referral=dvpita@gmail.com&idSignature=22>

_______________________________________
Best Regards,

Jose Pita
linkedin.com/in/pitajose

**Comment by took3n** [comment]
does not work any more

**Comment by kilrizzy** [comment]
Yep, looks like they pull an id now but not sure where it comes from :/

**Comment by Firkata** [comment]
I noticed that the stopped supporting this api:
https://files.coinmarketcap.com/static/img/coins/64x64/ethereum.png
Has anyone found another api to get icons for the coins?

**Comment by kilrizzy** [comment]
It seems each image is changed to an index value instead of coinmarketcaps api "id". However this index isn't listed in the API

If it helps anybody, here's what I used in my Laravel app to scan each token's page on coinmarketcap and grab the index value from the html / image url:

```

public function handle()
    {
        $tokens = \App\Token::whereNull('cmc_index')->get();
        foreach($tokens as $token){
            $this->getTokenIndex($token);
        }
    }

    private function getTokenIndex($token){
        $client = new \GuzzleHttp\Client();
        try {
            $request = $client->request('GET', 'https://coinmarketcap.com/currencies/' . $token->cmc_id . '/');
        }catch (\Exception $e){
            return null;
        }
        $response = $request->getBody()->getContents();
        $responseParts = explode('https://files.coinmarketcap.com/static/img/coins/32x32/',$response);
        if(isset($responseParts[1])){
            $responseParts2 = explode('.png',$responseParts[1]);
            if(!empty($responseParts2[0]) && is_numeric($responseParts2[0])){
                $tokenIndex = $responseParts2[0];
                $token->cmc_index = $tokenIndex;
                $token->save();
            }
        }
    }
```

Obviously it takes a while to run through each, but I just run this daily to store any new tokens that might be added

**Comment by ashutoshdev** [comment]
Hey guys ,
I was using url like the given below
https://files.coinmarketcap.com/static/img/coins/64x64/ethereum.png
now they have changed the images name like random no 1.png 2.png but they have not generated by the rank no they have generated with random no so any one can help me to find the images and graph images url..

Thanks

**Comment by ashutoshdev** [comment]
@kilrizzy thank you for reply ..

I am working in python but you have given best clue

thanks
